core/types/transaction_marshalling.go

```diff
@@ -60,6 +60,10 @@ func (tx *Transaction) MarshalJSON() ([]byte, error) {
 	// Other fields are set conditionally depending on tx type.
 	switch itx := tx.inner.(type) {
 	case *LegacyTx:
+		// Only derive the chain ID if tx is signed and post EIP-155
+		if (itx.V.Sign() != 0 || itx.R.Sign() != 0 || itx.S.Sign() != 0) && tx.ChainId().Sign() != 0 {
```

Comment:
```suggestion
		if itx.V != nil && isProtected(itx.V) {
```

This function is `core/types/transaction.go` and returns `true` if the input is considered a "protected" V value, per EIP-155. I think you can simplify it this way, because the `v`, `r`, and `s` values should be `nil` if the tx is not signed.

Reply:
Good point. I think we could simplify it further by just checking [`tx.Protected()`](https://github.com/ethereum/go-ethereum/blob/281e8cd5abaac86ed3f37f98250ff147b3c9fe62/core/types/transaction.go#LL253C1-L260C2)?

Reply:
Ah- yes, that would further simplify.

Reply:
I also think this should be after `enc.S` is set.